src/tests/test_inspect.py

```diff
@@ -127,3 +131,76 @@ def test_get_schema_names(ip):
     schema_names = inspect.get_schema_names()
     for schema in schema_names:
         assert schema in expected_schema_names
+
+
+@pytest.mark.parametrize(
+    "get_columns, rows, field_names",
+    [
+        [
+            [
+                {"column_a": "a", "column_b": "b"},
+                # the second row does not have column_b
+                {
+                    "column_a": "a2",
+                },
+            ],
+            [["a", "b"], ["a2", ""]],
+            ["column_a", "column_b"],
+        ],
+        [
+            [
+                {
+                    "column_a": "a2",
+                },
+                # contains an extra column
+                {"column_a": "a", "column_b": "b"},
+            ],
+            [["a2", ""], ["a", "b"]],
+            ["column_a", "column_b"],
+        ],
+        [
+            [
+                {"column_a": "a", "column_b": "b"},
+                {"column_b": "b2", "column_a": "a2"},
+            ],
+            [["a", "b"], ["a2", "b2"]],
+            ["column_a", "column_b"],
+        ],
+        [
+            [
+                dict(),
+                dict(),
+            ],
+            [[], []],
+            [],
+        ],
+        [
+            None,
+            [],
+            [],
+        ],
+    ],
+    ids=[
+        "missing-val-second-row",
+        "extra-val-second-row",
+        "keeps-order",
+        "empty-dictionaries",
+        "none-return-value",
+    ],
+)
+def test_columns_with_missing_values(ip, monkeypatch, get_columns, rows, field_names):
```

Comment:
I think we can add another test case when the schema is set

Reply:
added